### PR TITLE
Domain replication followup changes

### PR DESCRIPTION
**Pull request by meiliang86**
- Check host identity before processing domain replication tasks
- Add metrics for domain replication queue ack level

### CODE REVIEW

File: common/persistence/domainReplicationQueue.go [hunk 1]
```diff
@@ -51,8 +59,10 @@ func NewDomainReplicationQueue(queue Queue, logger log.Logger) DomainReplication
 
 type (
 	domainReplicationQueueImpl struct {
-		logger              log.Logger
 		queue               Queue
+		clusterName         string
```
Comment (wxing1292) [inline common/persistence/domainReplicationQueue.go]: current cluster name?

File: common/persistence/domainReplicationQueue.go [hunk 2]
```diff
@@ -137,6 +147,10 @@ func (q *domainReplicationQueueImpl) purgeAckedMessages() error {
 		return fmt.Errorf("failed to purge messages: %v", err)
 	}
 
+	q.metricsClient.
+		Scope(metrics.FrontendDomainReplicationQueueScope, metrics.SourceClusterTag(q.clusterName)).
```
Comment (wxing1292) [inline common/persistence/domainReplicationQueue.go]: i believe the source cluster tag is not necessary

File: common/persistence/domainReplicationQueue.go (hunk 1 above)
Comment (meiliang86) [reply common/persistence/domainReplicationQueue.go]: It's called clusterName everywhere else so I'm going to keep it as is.

File: common/persistence/domainReplicationQueue.go (hunk 2 above)
Comment (meiliang86) [reply common/persistence/domainReplicationQueue.go]: You are right. I'll remove.

**Comment by meiliang86** [comment]
#2393

**Pull request merged**